portal-bridge/src/bridge/beacon.rs

```diff
@@ -118,17 +136,19 @@ impl BeaconBridge {
         // If serving takes a little too long, then we want to serve the next one as soon as
         // possible, but not serve any extras until the following slot. "Skip" gets this behavior.
         interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+        interval.tick().await;
```

Comment:
I'm not sure why this is the case, but when I move the first ` interval.tick().await` call in the loop, the loop runs 2 times in a row at the beginning without waiting for the tick. I haven't dived into this but it seems that calling `interval.tick().await` first before the loop fixes this issue.

Reply:
This didn't make sense to me, so I checked the [documentation](https://dtantsur.github.io/rust-openstack/tokio/time/fn.interval.html). The first `tick` completes immediately.

So yes, if you just remove this line, it will execute the `loop` twice immediately. But if you do something like this:

```rust
let mut interval = interval(Duration::from_secs(12));
interval.set_missed_tick_behavior(MissedTickBehavior::Skip);

loop {
    interval.tick().await;

    let consensus_api = self.api.clone();
    let portal_client = self.portal_client.clone();
    Self::serve_latest(...).await;
}
```

I believe that you will have expected behavior.

But no big deal, up to you.